www/putific

```diff
@@ -603,26 +609,28 @@ if ($linkFile)
             "compression" => (isset($fmtMap[$cmp]) ? $fmtMap[$cmp] : false),
             "compressedprimary" => xcvt($l->compressedPrimary));
 
-        // Look for an existing link with the same URL.  If there's an
-        // existing link, we'll update it.  Otherwise we'll add this one
-        // as a new link.
-        for ($i = 0, $found = false ; $i < count($recLinks) ; $i++)
-        {
-            $recl = $recLinks[$i];
-            if (strcasecmp($recl["url"], $url) == 0)
+        if (!$replaceLinks) {
+            // Look for an existing link with the same URL.  If there's an
+            // existing link, we'll update it.  Otherwise we'll add this one
+            // as a new link.
+            for ($i = 0, $found = false ; $i < count($recLinks) ; $i++)
```

Comment:
Instead of this nested loop, I'd populate an array above the top loop with lowercased versions of the original URL pointing at their `rec link` objects.

Then the lookup is just an O(1) check if the lowercased new URL is in the array. (Not that speed matters in this case, but I think it's more readable)